src/playback/view/internal/msbasicpresetscategories.h

```diff
@@ -0,0 +1,521 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef MU_PLAYBACK_MSBASICPRESETSCATEGORIES_H
+#define MU_PLAYBACK_MSBASICPRESETSCATEGORIES_H
+
+#include <vector>
+
+#include "types/string.h"
+
+#include "midi/miditypes.h"
+
+namespace mu::playback {
+struct MsBasicItem {
+    String title;
+    std::vector<MsBasicItem> subItems;
+    midi::Program preset = {};
+
+    MsBasicItem(String&& title, std::vector<MsBasicItem>&& subItems)
+        : title(title), subItems(subItems) {}
+
+    MsBasicItem(midi::Program&& preset, String&& titleOverride = {})
+        : title(titleOverride), preset(preset) {}
+};
+
+// Based on https://docs.google.com/spreadsheets/d/1SwqZb8lq5rfv5regPSA10drWjUAoi65EuMoYtG-4k5s/edit#gid=1502434630
+static const std::vector<MsBasicItem> msBasicPresetCategories {
```

Comment:
maybe it's better to put it in a JSON file and read it when the program starts

Reply:
Reading from a file would be good because at some point we want to allow users to create custom categories for other SoundFonts besides MS Basic.

Reply:
Casper's solution also has its advantages: we won't need to read this info from the file (which is obviously faster and safer). I think we can keep this solution for now and improve it later if the need arises

Reply:
I also seem to recall that Vasily has a preference for putting things statically in code rather than reading from files, for performance reasons (but I believe that was actually in the context of `instruments.xml`)

Reply:
It's certainly quicker and easier to load from the source code in this case.

In the case of instruments, I never understood the arguments since we are still reading from `instruments.xml` anyway, but I would be keen to learn. Perhaps it's because some of the data is shared with the Muse Sounds guys and Vasily didn't want them to have to write a parser?

Of course we could always use the spreadsheet to generate C++ code rather than an XML document and then we would get the best of both worlds.